### PR TITLE
Enhancement: Detect when valid partkeys represent an offline or closed account

### PR DESCRIPTION
**Pull request by AlgoStephenAkiki**
Resolves #2060

Resolves all scenarios described below:

Scenario 1 (S1): Address is closed, but has a participation key that is still registered and valid given the current round number.
Scenario 2 (S2): Address is offline. Could imply no registration txn or an expired one.
Scenario 3 (S3): Address is online, with a valid participation key.

The breakdown of the scenarios by count is as follows:

S1: 7 addresses
S2: 43 addresses
S3: 2 addresses

The following actions are recommended for each scenario:

S1

Check that account is closed for the on-chain data. If account is closed, silently ignore since this account has been officially closed.

S2

Check that account is offline. If account is offline, return error indicating that no registration transaction has been issued or that previous registration transaction is expired.

S3

Recommend that error message indicates that user should regenerate participation key for the node.

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -1364,11 +1378,28 @@ func (node *AlgorandFullNode) VotingKeys(votingRound, keysRound basics.Round) []
...
-			node.log.Warnf("node.VotingKeys: Account %v not participating on round %d: on chain voting key differ from participation voting key for round %d", mismatchingAddr, votingRound, keysRound)
+
+			logStr := fmt.Sprintf("node.VotingKeys: Account %v not participating on round %d: on chain voting key differ from participation voting key for round %d", mismatchingAddr, votingRound, keysRound)
+
+			// If we are closed, upgrade this to info so we don't spam telemetry reporting
+			if warningFlags&bitAccountIsClosed == bitAccountIsClosed {
+				node.log.Info(logStr)
+			} else {
+				node.log.Warnf(logStr)
+			}
```
[the remark this answers is not in the document]
Comment (AlgoStephenAkiki) [reply node/node.go:1389]: This is just for scenario 1 where the account has been closed.  We agreed that we would simply upgrade the log message to info here to stop spam.  In scenario 2 & 3 we would have a different log message.

**Review by winder — DISMISSED** [review]

**Comment by AlgoStephenAkiki** [comment]
Updated PR to include all scenario warnings.

**Comment by cce** [comment]
No tests, was there manual testing to identify the logging working for S1/S2/S3?

**Comment by AlgoStephenAkiki** [comment]
@cce  Could you look at the updated PR?

File: node/node_test.go [hunk 2]
```diff
@@ -553,3 +553,22 @@ func TestAsyncRecord(t *testing.T) {
 	require.Equal(t, 10000, int(records[0].LastVote))
 	require.Equal(t, 20000, int(records[0].LastBlockProposal))
 }
+
+// OfflineOnlineClosedBitStatusTest a test that validates that the correct bits are being set
+func OfflineOnlineClosedBitStatusTest(t *testing.T) {
```
Comment (cce) [inline node/node_test.go:558]: This method doesn't start with Test, so it is not run by go test

File: node/node.go [hunk 3]
```diff
@@ -1363,12 +1390,33 @@ func (node *AlgorandFullNode) VotingKeys(votingRound, keysRound basics.Round) []
 		if matchingAccountsKeys[mismatchingAddr] {
 			continue
 		}
-		if warningFlags&bitMismatchingVotingKey == bitMismatchingVotingKey {
-			node.log.Warnf("node.VotingKeys: Account %v not participating on round %d: on chain voting key differ from participation voting key for round %d", mismatchingAddr, votingRound, keysRound)
+		if warningFlags&bitMismatchingVotingKey != 0 {
```
Comment (cce) [inline node/node.go:1393]: it seems like you have unified the logging cases between bitMismatchingVotingKey and bitMismatchingSelectionKey, so you could just change this to be `warningFlags&bitMismatchingVotingKey != 0 || warningFlags&bitMismatchingSelectionKey != 0`?

**Review by cce — CHANGES_REQUESTED** [review]
Need to rename test, would be nice to consolidate duplication

**Review by winder — APPROVED** [review]

**Pull request merged**